src/test/java/com/example/bookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,329 @@
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.math.BigDecimal;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class BookControllerTest {
+    protected static MockMvc mockMvc;
+    private static final Long EXISTING_ID = 1L;
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+    }
+
+    @WithMockUser(username = "admin", roles = {"ADMIN"})
+    @Test
+    @Sql(scripts = {
+            "classpath:database/fill-books-table.sql",
+            "classpath:database/fill-categories-table.sql",
+            "classpath:database/fill-books-categories-table.sql"
+    }, executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+    @Sql(scripts = {
+            "classpath:database/delete-from-books-categories-table.sql",
+            "classpath:database/delete-from-books-table.sql",
+            "classpath:database/delete-from-categories-table.sql"
+    }, executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+    @DisplayName(
+            "Verify that create book method is working correctly"
+    )
+    public void createBook_ValidCreateBookRequest_ReturnsBookDto() throws Exception {
+        CreateBookRequestDto createBookRequestDto = getCreateBookRequestDto();
+        String jsonRequest = objectMapper.writeValueAsString(createBookRequestDto);
+
+        MvcResult result = mockMvc.perform(post("/books")
+                        .content(jsonRequest)
+                        .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isCreated())
+                .andReturn();
+
+        BookDto expected = getCreatedBookDto();
+        BookDto actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(), BookDto.class);
+        EqualsBuilder.reflectionEquals(expected, actual, "id");
```

Comment:
use static imports
check and fix all places
when you are using reflectionEquals, add assertEquals ahead
```suggestion
        assertEquals(reflectionEquals(expected, actual, "id"));
```